### PR TITLE
Program to remove duplicates from a string

### PR DESCRIPTION
**Pull request by pankaz20**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### Description

We traverse through all the characters and we will check if their duplicate is present or not.If it is present then we will delete it.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#3894

### CODE REVIEW

**Comment by pankaz20** [comment]
@siddharth25pandey   Please review it.

**Review by Siddhant-K-code — APPROVED** [review]
LGTM!

**Comment by pankaz20** [comment]
@HarshCasper please Review it.

**Comment by Siddhant-K-code** [comment]
@pankaz20, It will be good if you can add a sample test case ( example ) to test/run it.

**Comment by pankaz20** [comment]
> @pankaz20, It will be good if you can add a sample test case ( example ) to test/run it.

Done

File: C-Plus-Plus/ds/duplicate_removal.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		str[index++] = str[i];
+}
+
+return str;
+}
+
+// Driver code
+int main()
+{
+char str[]= "warofwarrior";
```
Comment (Siddhant-K-code) [inline C-Plus-Plus/ds/duplicate_removal.cpp]: Make `str` for the user inputs

**Review by Siddhant-K-code — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/ds/duplicate_removal.cpp (hunk 1 above)
Comment (pankaz20) [reply C-Plus-Plus/ds/duplicate_removal.cpp]: Done

**Review by Siddhant-K-code — APPROVED** [review]
LGTM!

File: C-Plus-Plus/ds/duplicate_removal.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
+// CPP program to remove duplicate character
+// from character array and print in sorted
+// order
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/duplicate_removal.cpp]: Add a detailed description

File: C-Plus-Plus/ds/duplicate_removal.cpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+// Driver code
+int main()
+{
+char str[1000];
+cin>>str;
+int n = sizeof(str) / sizeof(str[0]);
+cout << removeDuplicate(str, n);
+return 0;
+}
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/duplicate_removal.cpp]: Indent the Code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/ds/duplicate_removal.cpp (hunk 3 above)
Comment (pankaz20) [reply C-Plus-Plus/ds/duplicate_removal.cpp]: Done

File: C-Plus-Plus/ds/duplicate_removal.cpp (hunk 2 above)
Comment (pankaz20) [reply C-Plus-Plus/ds/duplicate_removal.cpp]: Done

File: C-Plus-Plus/ds/duplicate_removal.cpp (hunk 3 above)
Comment (Siddhant-K-code) [reply C-Plus-Plus/ds/duplicate_removal.cpp]: @pankaz20 I think you should format your CPP code with clang-format or beautifier or an online formatter. it is for better readability.

Comment (pankaz20) [reply C-Plus-Plus/ds/duplicate_removal.cpp]: Done,Is it Ok now?

**Comment by siddharth25pandey** [comment]
@pankaz20 Code plagiarized from [here](https://www.geeksforgeeks.org/remove-duplicates-from-a-given-string/)

**Pull request closed without merging**